File: trunk/src/dotnet/yaxgl/YaxglLibrary/Container/Group.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;
namespace de.yaxgl
{
    //TODO: we need an other name for group
    public class Group : Container,Containable
    {
        
        public Group(string xmlfile,Container owner,string ID)
        {
            this.owner = owner;
            this.ID = ID;
            this.control = new System.Windows.Forms.Panel();

            XmlElement rootElement = null;

            rootElement = validateXmlDocument(xmlfile);

            parseXML(rootElement);
            
            //TODO:register GroupEvents
            
        }

        public override void initializeNativeControl(XmlElement xmlElement)
        {
            Location location = new Location(Convert.ToInt32(xmlElement.Attributes["xpos"].InnerText),
                                 Convert.ToInt32(xmlElement.Attributes["ypos"].InnerText));
            setLocation(location);

        }

        public override void notifyEvent(Component control, EventArgs eventArgs)
        {
            owner.notifyEvent(control, eventArgs);    
        }

        /* setting border to group true=border false=noneborder*/
        public void setBorder(bool border)
        {
            if (border)
            {
                ((System.Windows.Forms.Panel)this.control).BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
            }
            else
            {
                ((System.Windows.Forms.Panel)this.control).BorderStyle = System.Windows.Forms.BorderStyle.None;
            }
        
        }
    
    }
}

File: trunk/src/java/YaxglLibrary/de/yaxgl/Container/Group.java

package de.yaxgl.Container;

import org.eclipse.swt.SWT;
import org.w3c.dom.Element;

import de.yaxgl.Base.*;
import de.yaxgl.EventDispatcher.*;
import de.yaxgl.Helper.*;


@SuppressWarnings("serial")
public class Group extends Container implements Containable {

	public Group(String xmlfile,Container owner,String ID)
	{
		this.owner = owner;
		this.ID=ID;
		org.eclipse.swt.widgets.Composite composite=(org.eclipse.swt.widgets.Composite)this.owner.getNativeComponent();                
        this.control = new org.eclipse.swt.widgets.Composite(composite,SWT.NONE); 
        Element rootElement = validateXmlDocument(xmlfile);
        try {
			parseXML(rootElement);
		} catch (Exception e) {
			System.out.println(e.getMessage());
		}
	}
	
	
	@Override
	public void initializeNativeControl(Element xmlElement) {
		Location location=new Location(Integer.valueOf(xmlElement.getAttribute("xpos")),Integer.valueOf(xmlElement.getAttribute("ypos")));
        setLocation(location);
       
	}
	
	@Override
	public void notifyEvent(Component control, EventArgs eventArgs) {
		this.owner.notifyEvent(control, eventArgs);
	}
	

	

}
